akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/PersistentBehaviorImpl.scala

```diff
@@ -45,7 +45,7 @@ private[akka] final case class PersistentBehaviorImpl[Command, Event, State](
           recovery,
           holdingRecoveryPermit = false,
           settings = settings,
-          internalStash = StashBuffer(settings.stashCapacity)
+          internalStash = stash(settings)
```

Comment:
ok, let's see how we can add the `PostStop` cleanup.
I don't think we can use `Behaviors.receiveSignal`, because that will consume the signal and not propagate to inner behaviors that might also need it.

Perhaps via `Behaviors.intercept`.

```
Behaviors.setup[EventsourcedBehavior.InternalProtocol] { ctx ⇒
  Behaviors.withTimers { timers ⇒
    Behaviors.intercept(...
```

Reply:
I guess you meant `Behaviors.tap`? Yeah, makes sense. Will add changes to this PR.

Reply:
Done